### PR TITLE
Adding snippet names

### PR DESCRIPTION
**Pull request by idk23535**
---
title: Issue # |Snippet Names
---

Discord Username: @idk23535

## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description

Added snippet names to customize snippets more! The snippet names just create more uniqueness of snippets. I also added snippet names to your dashboard.

## Related Tickets & Documents

- Related Issue #
- Closes #

## Added/updated tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

This is what it looks like 

Screenshot 2023-08-05 3 01 29 AM
Screenshot 2023-08-05 2 58 16 AM
Screenshot 2023-08-05 2 59 11 AM

### CODE REVIEW

File: packages/app/src/app/race/_components/race/race-multiplayer.tsx [hunk 1]
```diff
@@ -113,6 +114,12 @@ export default function RaceMultiplayer({
       setSnippet(snippet);
       setRaceId(race.id);
       setParticipantId(raceParticipantId);
+
+      if (snippet.name === "undefined") {
+        setSnippetTitle("Type this code");
+      } else {
+        setSnippetTitle(snippet.name!)
```
Comment (scape76) [inline packages/app/src/app/race/_components/race/race-multiplayer.tsx:124]: why do you need separate useState for that? just say const snippetTitle = snippet.name ? "..." : "..."

Comment (idk23535) [reply packages/app/src/app/race/_components/race/race-multiplayer.tsx:124]: yeah, I forgot about that i'll fix it.

File: packages/app/prisma/schema.prisma [hunk 2]
```diff
@@ -117,6 +117,7 @@ model Achievement {
 
 model Snippet {
     id String @id @default(cuid())
+    name String? @default("undefined")
```
Comment (scape76) [inline packages/app/prisma/schema.prisma:120]: you don't need a default value here, as it's an optional field

Comment (idk23535) [reply packages/app/prisma/schema.prisma:120]: ok

**Pull request merged**